ssl/quic/quic_port.c

```diff
@@ -1401,6 +1403,8 @@ static void generate_new_token(QUIC_CHANNEL *ch, BIO_ADDR *peer)
         return;
     }
 
+    memset(&token, 0, sizeof(QUIC_VALIDATION_TOKEN));
```

Comment:
I still don't think this is enough because similar to what @vdukhovni commented, `generate_token` in the future might overwrite `token.remote_addr` with invalid data before returning an error. So probably `generate_token` should be a separate `if` case that doesn't run `cleanup_validation_token` on error.

Reply:
I'm sorry, but this seems like concern over a future possible change that we would need to take into account at that point in time anyway.

As it currently stands, we set token.remote_addr to NULL prior to calling generate_token().  generte_token() then during its execution allocates remote_addr and writes to it via BIO_ADDR_rawaddress().  If either of those fail, we call cleanup_validation_token, where remote_addr in the struct will either be NULL, in which case the free therein is a no-op, or a valid pointer, in which case the free would work as you would expect.  The contents of the pointer that remote_addr points to is irrelevant to the free operation.

Can you describe the sort of scenario in which whats being done might have a negative impact on a future change given the above?

Reply:
I guess it’s ok, just trying to prevent future problems. We can leave it as is

Reply:
Please help me understand what you're looking for here to improve, because I honestly don't.  We make use of this pattern frequently.  A stack allocated structure with a memset to pre-set all its values is no different than what we do in (for example) SSL_CTX_new_ex, SSL_new, ossl_quic_channel_alloc, etc, in which we allocate a structure with OPENSSL_zalloc, to ensure that all pointer values are set to NULL, so that freeing operations in the event of an error are either legitimate free operations, or no-ops.

Whats different about this use case that exposes us to risk of future changes?

Reply:
I agree that it's concern over a future possible change that we would need to take into account at that time anyways, so the code is fine as-is